File: headless_solutions/n_repeated_element_in_size_2n_array/solution.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var repeatedNTimes = function(nums) {
    let count = {};
    for (let num of nums) {
        count[num] = (count[num] || 0) + 1;
    }
    for (let num in count) {
        if (count[num] > 1) {
            return parseInt(num);
        }
    }
    return -1; // This should never happen
};

File: headless_solutions/n_repeated_element_in_size_2n_array/solution.cpp
class Solution {
public:
    int repeatedNTimes(vector<int>& nums) {
        unordered_map<int, int> count;
        for (int num : nums) {
            count[num]++;
        }
        for (auto& pair : count) {
            if (pair.second > 1) {
                return pair.first;
            }
        }
        return -1; // This should never happen
    }
};
